addons/binding/org.openhab.binding.elerotransmitterstick/src/main/java/org/openhab/binding/elerotransmitterstick/stick/TransmitterStick.java

```diff
@@ -0,0 +1,390 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Volker Bier - Initial contribution
+ */
+public class TransmitterStick {
+    private final Logger logger = LoggerFactory.getLogger(TransmitterStick.class);
+
+    private final CommandWorker worker;
+
+    private final HashMap<Integer, ArrayList<StatusListener>> allListeners = new HashMap<>();
+
+    private final StickListener listener;
+
+    private EleroTransmitterStickConfig config;
+
+    public TransmitterStick(EleroTransmitterStickConfig stickConfig, StickListener l) {
+        config = stickConfig;
+        worker = new CommandWorker();
+
+        listener = l;
+    }
+
+    public synchronized void initialize() {
+        logger.debug("Initializing Transmitter Stick...");
+        worker.start();
+        logger.debug("Transmitter Stick initialized, worker running.");
+    }
+
+    public synchronized void dispose() {
+        logger.debug("Disposing Transmitter Stick...");
+        allListeners.clear();
+        worker.terminateUpdates();
+        logger.debug("Transmitter Stick disposed.");
+    }
+
+    public int[] getKnownIds() {
+        return worker.knownIds;
+    }
+
+    public void sendCommand(CommandType cmd, int... channelIds) {
+        worker.executeCommand(cmd, channelIds);
+    }
+
+    public void requestUpdate(int... channelIds) {
+        worker.requestUpdate(channelIds);
+    }
+
+    public void addStatusListener(int channelId, StatusListener listener) {
+        synchronized (allListeners) {
+            ArrayList<StatusListener> listeners = allListeners.get(channelId);
+            if (listeners == null) {
+                listeners = new ArrayList<>();
+                allListeners.put(channelId, listeners);
+            }
+            listeners.add(listener);
+        }
+    }
+
+    public void removeStatusListener(int channelId, StatusListener listener) {
+        synchronized (allListeners) {
+            ArrayList<StatusListener> listeners = allListeners.get(channelId);
+            if (listeners != null) {
+                listeners.remove(listener);
+
+                if (listeners.isEmpty()) {
+                    allListeners.remove(channelId);
+                }
+            }
+        }
+    }
+
+    private void notifyListeners(int channelId, ResponseStatus status) {
+        synchronized (allListeners) {
+            ArrayList<StatusListener> listeners = allListeners.get(channelId);
+            if (listeners != null) {
+                for (StatusListener l : listeners) {
+                    l.statusChanged(channelId, status);
+                }
+            }
+        }
+    }
+
+    /**
+     * Make sure we have
+     * - only one INFO for the same channel ids
+     * - only one other command for the same channel ids
+     */
+    private static boolean prepareAddition(Command newCmd, Collection<Command> coll) {
+        Iterator<Command> queuedCommands = coll.iterator();
+        while (queuedCommands.hasNext()) {
```

Comment:
Please write it as: `for(Command existingCmd : coll)`, this saves 2 lines and is more readable.

Reply:
No, that does not work, as I use the iterators remove method.